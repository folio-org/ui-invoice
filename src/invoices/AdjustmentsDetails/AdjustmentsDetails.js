import React from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import {
  Col,
  Row,
} from '@folio/stripes/components';
import {
  AmountWithCurrencyField,
  FrontendSortingMCL,
} from '@folio/stripes-acq-components';

import {
  ADJUSTMENT_PRORATE_LABELS,
  ADJUSTMENT_RELATION_TO_TOTAL_LABELS,
  ADJUSTMENT_TYPE_VALUES,
} from '../../common/constants';

const sortedColumn = 'description';
const visibleColumns = ['description', 'value', 'prorate', 'relationToTotal'];
const columnMapping = {
  description: <FormattedMessage id="ui-invoice.adjustment.description" />,
  value: <FormattedMessage id="ui-invoice.adjustment.amount" />,
  prorate: <FormattedMessage id="ui-invoice.settings.adjustments.prorate" />,
  relationToTotal: <FormattedMessage id="ui-invoice.settings.adjustments.relationToTotal" />,
};

const SORTERS = { description: ({ description }) => description?.toLowerCase() };

const AdjustmentsDetails = ({ adjustments, currency }) => {
  const resultsFormatter = {
    value: d => (
      d.type === ADJUSTMENT_TYPE_VALUES.amount
        ? (
          <AmountWithCurrencyField
            amount={d.value}
            currency={currency}
          />
        )
        : `${d.value}%`
    ),
    prorate: d => d.prorate && <FormattedMessage id={ADJUSTMENT_PRORATE_LABELS[d.prorate]} />,
    relationToTotal: d => <FormattedMessage id={ADJUSTMENT_RELATION_TO_TOTAL_LABELS[d.relationToTotal]} />,
  };

  return (
    <Row>
      <Col xs={12}>
        <FrontendSortingMCL
          columnMapping={columnMapping}
          contentData={adjustments}
          formatter={resultsFormatter}
          id="invoice-lines-adjustments-list"
          sortedColumn={sortedColumn}
          sorters={SORTERS}
          visibleColumns={visibleColumns}
          columnIdPrefix="adjustments"
        />
      </Col>
    </Row>
  );
};

AdjustmentsDetails.propTypes = {
  adjustments: PropTypes.arrayOf(PropTypes.object),
  currency: PropTypes.string,
};

export default AdjustmentsDetails;
