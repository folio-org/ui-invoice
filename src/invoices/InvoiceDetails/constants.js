export const VENDOR_STATUS = {
  ACTIVE: 'Active',
  INACTIVE: 'Inactive',
  PENDING: 'Pending',
};

export const ACQ_ERROR_TYPE = {
  order: 'order',
};

export const INVOICE_OMITTED_FIELDS = [
  'approvalDate',
  'approvedBy',
  'documents',
  'fiscalYearId',
  'id',
  'links',
  'metadata',
  'paymentDate',
  'status',
  'voucherNumber',
];
export const PO_LINE_PAYMENT_STATUS = {
  NO_CHANGE: 'noChange',
  AWAITING_PAYMENT: 'awaitingPayment',
  PARTIALLY_PAID: 'partiallyPaid',
  FULLY_PAID: 'fullyPaid',
  CANCELLED: 'cancelled',
};

export const PO_LINE_PAYMENT_STATUSES = [
  PO_LINE_PAYMENT_STATUS.NO_CHANGE,
  PO_LINE_PAYMENT_STATUS.AWAITING_PAYMENT,
  PO_LINE_PAYMENT_STATUS.PARTIALLY_PAID,
  PO_LINE_PAYMENT_STATUS.FULLY_PAID,
  PO_LINE_PAYMENT_STATUS.CANCELLED,
];

export const PO_LINE_PAYMENT_STATUS_LABELS = {
  [PO_LINE_PAYMENT_STATUS.NO_CHANGE]: 'No Change',
  [PO_LINE_PAYMENT_STATUS.AWAITING_PAYMENT]: 'Awaiting Payment',
  [PO_LINE_PAYMENT_STATUS.PARTIALLY_PAID]: 'Partially Paid',
  [PO_LINE_PAYMENT_STATUS.FULLY_PAID]: 'Fully Paid',
  [PO_LINE_PAYMENT_STATUS.CANCELLED]: 'Cancelled',
};

export const ORDER_TYPE = {
  ONE_TIME: 'One-Time',
  ONGOING: 'Ongoing',
};
