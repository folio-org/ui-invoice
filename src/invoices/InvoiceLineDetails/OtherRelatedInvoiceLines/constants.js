import { FormattedMessage } from 'react-intl';

export const COLUMN_INVOICE_DATE = 'invoiceDate';

export const SORTABLE_FIELDS = [COLUMN_INVOICE_DATE];
export const DEFAULT_SORT_FIELD = COLUMN_INVOICE_DATE;
export const VISIBLE_COLUMNS = [
  'vendorInvoiceNo',
  'invoiceLine',
  'fiscalYear',
  COLUMN_INVOICE_DATE,
  'vendorCode',
  'subscriptionStart',
  'subscriptionEnd',
  'subscriptionDescription',
  'status',
  'quantity',
  'amount',
  'comment',
];

export const COLUMN_MAPPING = {
  invoiceLine: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.invoiceLine" />,
  fiscalYear: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.fiscalYear" />,
  [COLUMN_INVOICE_DATE]: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.invoiceDate" />,
  vendorCode: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.vendorCode" />,
  subscriptionStart: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.subscriptionStart" />,
  subscriptionEnd: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.subscriptionEnd" />,
  subscriptionDescription: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.subscriptionDescription" />,
  vendorInvoiceNo: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.vendorInvoiceNo" />,
  status: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.status" />,
  quantity: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.quantity" />,
  amount: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.amount" />,
  comment: <FormattedMessage id="ui-invoice.otherRelatedInvoiceLines.comment" />,
};
