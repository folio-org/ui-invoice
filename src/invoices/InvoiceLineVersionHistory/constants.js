export const INVOICE_LINE_FIELDS_LABEL_MAP = {
  'accountingCode': 'ui-invoice.invoice.accountingCode',
  'accountNumber': 'ui-invoice.invoice.accountNumber',
  'adjustments[\\d].description': 'ui-invoice.settings.adjustments.description',
  'adjustments[\\d].exportToAccounting': 'ui-invoice.settings.adjustments.exportToAccounting',
  'adjustments[\\d].id': 'ui-invoice.invoice.details.lines.list.adjustments',
  'adjustments[\\d].prorate': 'ui-invoice.settings.adjustments.prorate',
  'adjustments[\\d].relationToTotal': 'ui-invoice.settings.adjustments.relationToTotal',
  'adjustments[\\d].type': 'ui-invoice.settings.adjustments.type',
  'adjustments[\\d].value': 'ui-invoice.settings.adjustments.value',
  'adjustmentsTotal': 'ui-invoice.invoice.details.information.adjustment',
  'comment': 'ui-invoice.invoiceLine.comment',
  'description': 'ui-invoice.invoiceLine.description',
  'fundDistributions[\\d].id': 'stripes-acq-components.fundDistribution.id',
  'fundDistributions[\\d].amount': 'stripes-acq-components.fundDistribution.amount',
  'fundDistributions[\\d].code': 'ui-invoice.invoice.details.lines.list.fundCode',
  'fundDistributions[\\d].distributionType': 'stripes-acq-components.fundDistribution.type',
  'fundDistributions[\\d].fundId': 'ui-invoice.invoiceLine.fundId',
  'fundDistributions[\\d].value': 'stripes-acq-components.fundDistribution.value',
  'fundDistributions[\\d].expenseClassId': 'stripes-acq-components.fundDistribution.expenseClass',
  'fundDistributions[\\d].encumbrance': 'ui-invoice.invoiceLine.releaseEncumbrance',
  'invoiceId': 'ui-invoice.invoiceLine.invoiceId',
  'invoiceLineNumber': 'ui-invoice.invoice.invoiceLineNumber',
  'invoiceLineStatus': 'ui-invoice.invoiceLine.invoiceLineStatus',
  'poLineId': 'ui-invoice.invoiceLine.poLineId',
  'quantity': 'ui-invoice.invoiceLine.quantity',
  'referenceNumbers[\\d].refNumber': 'ui-invoice.invoice.details.lines.list.vendorRefNumber',
  'referenceNumbers[\\d].refNumberType': 'stripes-acq-components.referenceNumbers.refNumberType',
  'releaseEncumbrance': 'ui-invoice.invoiceLine.releaseEncumbrance',
  'subscriptionEnd': 'ui-invoice.invoiceLine.subscriptionEnd',
  'subscriptionInfo': 'ui-invoice.invoiceLine.subscriptionInfo',
  'subscriptionStart': 'ui-invoice.invoiceLine.subscriptionStart',
  'subTotal': 'ui-invoice.invoice.details.lines.list.subTotal',
  'total': 'ui-invoice.invoice.details.lines.list.total',
};

export const HIDDEN_INVOICE_LINE_FIELDS = [];

export const FUND_DISTRIBUTION_VISIBLE_COLUMNS = [
  'name',
  'expenseClass',
  'value',
  'amount',
];
