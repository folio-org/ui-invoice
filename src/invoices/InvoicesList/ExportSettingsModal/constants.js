export const EXPORT_INVOICE_FIELDS = {
  vendorInvoiceNo: 'Vendor invoice number',
  vendorCode: 'Vendor code',
  status: 'Status',
  invoiceDate: 'Invoice date',
  fiscalYear: 'Fiscal year',
  paymentDue: 'Payment due',
  approvalDate: 'Approved date',
  approvedBy: 'ApprovedBy',
  paymentTerms: 'Terms',
  acqUnitIds: 'Acquisitions units',
  note: 'Note',
  billTo: 'Bill to',
  batchGroup: 'Batch group',
  paymentDate: 'Payment date',
  totalUnits: 'Total units',
  subTotal: 'Sub total',
  adjustmentsTotal: 'Total adjustments',
  totalAmount: 'Total amount',
  lockTotal: 'Lock total amount',
  invoiceFundDistributions: 'Invoice fund distributions',
  invoiceAdjustments: 'Invoice adjustments',
  accountingCode: 'Accounting code',
  vendorAddress: 'Vendor address',
  paymentMethod: 'Payment method',
  chkSubscriptionOverlap: 'Check subscription overlap',
  exportToAccounting: 'Export to accounting',
  enclosureNeeded: 'Enclosure needed',
  currency: 'Currency',
  exchangeRate: 'Exchange rate',
  invoiceTags: 'Invoice tags',
  createdBy: 'Created by',
  dateCreated: 'Date created',
  updatedBy: 'Updated by',
  dateUpdated: 'Date updated',
};

export const EXPORT_INVOICE_LINE_FIELDS = {
  invoiceLineNumber: 'Invoice line number',
  description: 'Description (Title)',
  poLineNumber: 'POLine number',
  subscriptionInfo: 'Subscription info',
  subscriptionStart: 'Subscription start',
  subscriptionEnd: 'Subscription end',
  comment: 'Comment',
  accountNumber: 'Account number',
  lineAccountingCode: 'Accounting code',
  quantity: 'Quantity',
  lineSubTotal: 'Sub total',
  lineAdjustments: 'Invoice line adjustments',
  total: 'Total',
  lineFundDistributions: 'Invoice line fund distributions',
  externalAccountNumber: 'External account number',
  referenceNumbers: 'Vendor reference number, reference type',
  lineTags: 'Invoice line tags',
  voucherNumber: 'Voucher number',
  voucherStatus: 'Voucher status',
  voucherDate: 'Voucher date',
  disbursementNumber: 'Disbursement number',
  disbursementDate: 'Disbursement date',
  invoiceLineCreatedBy: 'Created by (Invoice Line)',
  invoiceLineDateCreated: 'Date created (Invoice Line)',
  invoiceLineUpdatedBy: 'Updated by (Invoice Line)',
  invoiceLineDateUpdated: 'Date updated (Invoice Line)',
};

export const EXPORT_INVOICE_FIELDS_OPTIONS = Object.keys(EXPORT_INVOICE_FIELDS).map(field => ({
  label: EXPORT_INVOICE_FIELDS[field],
  value: field,
}));

export const EXPORT_INVOICE_LINE_FIELDS_OPTIONS = Object.keys(EXPORT_INVOICE_LINE_FIELDS).map(field => ({
  label: EXPORT_INVOICE_LINE_FIELDS[field],
  value: field,
}));
