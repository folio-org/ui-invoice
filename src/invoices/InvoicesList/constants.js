// eslint-disable-next-line import/prefer-default-export
export const FILTERS = {
  PAYMENT_METHOD: 'paymentMethod',
  STATUS: 'status',
  VENDOR: 'vendorId',
  DATE_CREATED: 'createdDate',
  INVOICE_DATE: 'invoiceDate',
  PAYMENT_DUE: 'paymentDue',
  APPROVAL_DATE: 'approvalDate',
  SOURCE: 'source',
  EXPORT_TO_ACCOUNTING: 'exportToAccounting',
  ACQUISITIONS_UNIT: 'acquisitionsUnit',
  TAGS: 'tags',
};
