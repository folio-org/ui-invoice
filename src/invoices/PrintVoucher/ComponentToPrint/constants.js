import React from 'react';
import { FormattedMessage } from 'react-intl';

export const LINE_FIELDS_MAP = {
  poLineNumber: 'poLineNumber',
  poLineDescription: 'poLineDescription',
  quantityPhysical: 'cost.quantityPhysical',
  quantityElectronic: 'cost.quantityElectronic',
  currency: 'cost.currency',
  listUnitPrice: 'cost.listUnitPrice',
  listUnitPriceElectronic: 'cost.listUnitPriceElectronic',
  additionalCost: 'cost.additionalCost',
  discount: 'cost.discount',
  poLineEstimatedPrice: 'cost.poLineEstimatedPrice',
  fundDistribution: 'fundDistribution',
  productIds: 'productIdentifier',
  materialTypePhysical: 'materialType',
  materialTypeE: 'materialTypeE',
  acquisitionMethod: 'acquisitionMethod',
  titleOrPackage: 'titleOrPackage',
  publicationDate: 'publicationDate',
  publisher: 'publisher',
  rush: 'rush',
  vendorAccount: 'vendorDetail.vendorAccount',
  // instructions: 'vendorDetail.instructions',
};

export const LINE_FIELDS_LABELS = {
  [LINE_FIELDS_MAP.poLineNumber]: <FormattedMessage id="ui-orders.poLine.number" />,
  [LINE_FIELDS_MAP.poLineDescription]: <FormattedMessage id="ui-orders.poLine.poLineDescription" />,
  [LINE_FIELDS_MAP.quantityPhysical]: <FormattedMessage id="ui-orders.cost.quantityPhysical" />,
  [LINE_FIELDS_MAP.quantityElectronic]: <FormattedMessage id="ui-orders.cost.quantityElectronic" />,
  [LINE_FIELDS_MAP.currency]: <FormattedMessage id="ui-orders.cost.currency" />,
  [LINE_FIELDS_MAP.listUnitPrice]: <FormattedMessage id="ui-orders.cost.listPrice" />,
  [LINE_FIELDS_MAP.listUnitPriceElectronic]: <FormattedMessage id="ui-orders.cost.unitPriceOfElectronic" />,
  [LINE_FIELDS_MAP.additionalCost]: <FormattedMessage id="ui-orders.cost.additionalCost" />,
  [LINE_FIELDS_MAP.discount]: <FormattedMessage id="ui-orders.cost.discount" />,
  [LINE_FIELDS_MAP.poLineEstimatedPrice]: <FormattedMessage id="ui-orders.cost.estimatedPrice" />,
  [LINE_FIELDS_MAP.fundDistribution]: <FormattedMessage id="ui-orders.poLine.fundDistribution" />,
  [LINE_FIELDS_MAP.productIds]: <FormattedMessage id="ui-orders.itemDetails.productId" />,
  [LINE_FIELDS_MAP.materialTypePhysical]: <FormattedMessage id="ui-orders.filter.materialType.physical" />,
  [LINE_FIELDS_MAP.materialTypeE]: <FormattedMessage id="ui-orders.filter.materialType.electronic" />,
  [LINE_FIELDS_MAP.acquisitionMethod]: <FormattedMessage id="ui-orders.poLine.acquisitionMethod" />,
  [LINE_FIELDS_MAP.titleOrPackage]: <FormattedMessage id="ui-orders.orderLineList.titleOrPackage" />,
  [LINE_FIELDS_MAP.publicationDate]: <FormattedMessage id="ui-orders.itemDetails.publicationDate" />,
  [LINE_FIELDS_MAP.publisher]: <FormattedMessage id="ui-orders.itemDetails.publisher" />,
  [LINE_FIELDS_MAP.rush]: <FormattedMessage id="ui-orders.poLine.rush" />,
  [LINE_FIELDS_MAP.vendorAccount]: <FormattedMessage id="ui-orders.search.vendorDetail.vendorAccount" />,
  'vendorDetail.instructions': <FormattedMessage id="ui-orders.vendor.instructions" />,
};
