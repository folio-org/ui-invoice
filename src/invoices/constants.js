export const SECTIONS_INVOICE = {
  INFORMATION: 'information',
  LINES: 'invoiceLines',
  VENDOR_DETAILS: 'vendorDetails',
  VOUCHER: 'voucher',
  DOCUMENTS: 'documents',
  ADJUSTMENTS: 'adjustments',
};

export const SECTIONS_INVOICE_LINE = {
  invoiceLineInformation: 'invoiceLineInformation',
  fundDistribution: 'fundDistribution',
  adjustments: 'adjustments',
};

export const SECTIONS_VOUCHER = {
  VOUCHER: 'voucher',
  VOUCHER_LINES: 'voucherLines',
};
