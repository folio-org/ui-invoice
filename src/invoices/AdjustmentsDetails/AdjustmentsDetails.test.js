import { render } from '@folio/jest-config-stripes/testing-library/react';

import { adjustment } from '../../../test/jest/fixtures';
import { ADJUSTMENT_TYPE_VALUES } from '../../common/constants';
import AdjustmentsDetails from './AdjustmentsDetails';

const renderAdjustmentsDetails = (props) => (render(
  <AdjustmentsDetails {...props} />,
));

describe('AdjustmentsDetails', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should display adjustments table', () => {
    const { asFragment } = renderAdjustmentsDetails({
      currency: 'USD',
      adjustments: [adjustment, { ...adjustment, type: ADJUSTMENT_TYPE_VALUES.percent }],
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
