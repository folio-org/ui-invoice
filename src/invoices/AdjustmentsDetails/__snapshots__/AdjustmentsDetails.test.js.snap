// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdjustmentsDetails should display adjustments table 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-xs-12"
    >
      <div
        aria-live="assertive"
        aria-relevant="additions"
        class="sr-only"
        tabindex="-1"
      />
      <div
        aria-rowcount="3"
        class="mclContainer"
        data-total-count="0"
        id="invoice-lines-adjustments-list"
        role="grid"
        style="position: relative; overflow: hidden; height: auto; width: 100%;"
        tabindex="0"
      >
        <div
          class="mclHeaderContainer"
        >
          <div
            aria-rowindex="1"
            class="mclHeaderRow"
            role="row"
            style="display: flex;"
          >
            <div
              aria-sort="ascending"
              class="mclHeader mclClickable mclSorted mclAscending"
              id="adjustments-list-column-description"
              role="columnheader"
              style="flex: 0 0 auto;"
            >
              <div
                class="mclHeaderOuter"
                data-test-clickable-header="true"
                id="adjustments-clickable-list-column-description"
                role="button"
                tabindex="0"
              >
                <div
                  class="mclHeaderInner"
                >
                  <div
                    style="max-width: calc(100%-15px);"
                  >
                    ui-invoice.adjustment.description
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-sort="none"
              class="mclHeader"
              id="adjustments-list-column-value"
              role="columnheader"
              style="flex: 0 0 auto;"
            >
              ui-invoice.adjustment.value
            </div>
            <div
              aria-sort="none"
              class="mclHeader"
              id="adjustments-list-column-prorate"
              role="columnheader"
              style="flex: 0 0 auto;"
            >
              ui-invoice.settings.adjustments.prorate
            </div>
            <div
              aria-sort="none"
              class="mclHeader"
              id="adjustments-list-column-relationtototal"
              role="columnheader"
              style="flex: 0 0 auto;"
            >
              ui-invoice.settings.adjustments.relationToTotal
            </div>
          </div>
        </div>
        <div
          class="mclScrollable"
          style="position: static; width: 100%;"
          tabindex="0"
        >
          <div
            class="mclRowContainer"
            role="rowgroup"
            style="position: static;"
          >
            <div
              aria-rowindex="2"
              class="mclRowFormatterContainer"
              data-row-index="row-0"
              role="row"
              style="position: static;"
            >
              <div
                aria-label=""
                class="mclRow mclIsOdd"
                data-row-inner="0"
                style="min-width: 0px;"
                tabindex="0"
              >
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                />
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                >
                  $25.50
                </div>
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                />
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                >
                  ui-invoice.adjustments.relationToTotal.inAdditionTo
                </div>
              </div>
            </div>
            <div
              aria-rowindex="3"
              class="mclRowFormatterContainer"
              data-row-index="row-1"
              role="row"
              style="position: static;"
            >
              <div
                aria-label=""
                class="mclRow"
                data-row-inner="1"
                style="min-width: 0px;"
                tabindex="0"
              >
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                />
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                >
                  25.5%
                </div>
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                />
                <div
                  class="mclCell  mclCellStyle"
                  role="gridcell"
                >
                  ui-invoice.adjustments.relationToTotal.inAdditionTo
                </div>
              </div>
            </div>
            <div
              class="mclEndOfListContainer"
              data-end-of-list="2"
            >
              <div
                class="mclCenteredContainer"
                style="width: 100%; visibility: hidden; height: 0px; padding: 0px;"
              >
                <div
                  class="textCentered"
                  data-test-layout="true"
                >
                  <div
                    class="mclEndOfList"
                    data-test-layout="true"
                  >
                    <span>
                      Icon
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="textCentered noPaddingBottom"
        data-test-layout="true"
      >
        <div
          class="mclEndOfList"
          data-test-layout="true"
        >
          <span>
            Icon
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
